tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryCorrections/Test_DerivedProductCorrection.cpp

```diff
@@ -0,0 +1,272 @@
+          typename... PrimTags, typename... VolumeTags>
+struct ComputeBoundaryCorrectionHelperImpl<
+    DerivedCorrection, tmpl::list<PackagedFieldTags...>,
+    tmpl::list<EvolvedTags...>, tmpl::list<FluxTags...>,
+    tmpl::list<TempTags...>, tmpl::list<PrimTags...>,
+    tmpl::list<VolumeTags...>> {
+  template <typename PackagedVariables, typename EvolvedVariables,
+            typename FluxVariables, typename TempVariables,
+            typename PrimVariables, typename VolumeVariables>
+  static double dg_package_data(
+      const gsl::not_null<DerivedCorrection*> derived_correction,
+      const gsl::not_null<PackagedVariables*> packaged_variables,
+      const EvolvedVariables& evolved_variables,
+      const FluxVariables& flux_variables, const TempVariables& temp_variables,
+      const PrimVariables& prim_variables,
+      const VolumeVariables& volume_variables,
+      const tnsr::i<DataVector, 3, Frame::Inertial>& normal_covector,
+      const tnsr::I<DataVector, 3, Frame::Inertial>& normal_vector,
+      const std::optional<tnsr::I<DataVector, 3, Frame::Inertial>>&
+          mesh_velocity,
+      const std::optional<Scalar<DataVector>>&
+          normal_dot_mesh_velocity) noexcept {
+    return derived_correction->dg_package_data(
+        make_not_null(&get<PackagedFieldTags>(*packaged_variables))...,
+        get<EvolvedTags>(evolved_variables)...,
+        get<FluxTags>(flux_variables)..., get<TempTags>(temp_variables)...,
+        get<PrimTags>(prim_variables)..., normal_covector, normal_vector,
+        mesh_velocity, normal_dot_mesh_velocity,
+        get<VolumeTags>(volume_variables)...);
+  }
+
+  template <typename EvolvedVariables, typename PackagedVariables>
+  static void dg_boundary_terms(
+      const gsl::not_null<DerivedCorrection*> derived_correction,
+      const gsl::not_null<EvolvedVariables*> boundary_corrections,
+      const PackagedVariables& internal_packaged_fields,
+      const PackagedVariables& external_packaged_fields,
+      dg::Formulation dg_formulation) noexcept {
+    derived_correction->dg_boundary_terms(
+        make_not_null(&get<EvolvedTags>(*boundary_corrections))...,
+        get<PackagedFieldTags>(internal_packaged_fields)...,
+        get<PackagedFieldTags>(external_packaged_fields)..., dg_formulation);
+  }
+};
+
+template <typename DerivedCorrection, typename EvolvedTagList,
+          typename FluxTagList>
+using ComputeBoundaryCorrectionHelper = ComputeBoundaryCorrectionHelperImpl<
+    DerivedCorrection, typename DerivedCorrection::dg_package_field_tags,
+    EvolvedTagList, FluxTagList,
+    typename DerivedCorrection::dg_package_data_temporary_tags,
+    typename DerivedCorrection::dg_package_data_primitive_tags,
+    typename DerivedCorrection::dg_package_data_volume_tags>;
+
+using equation_of_state_type = EquationsOfState::IdealFluid<true>;
+
+template <typename DerivedGhCorrection, typename DerivedValenciaCorrection>
+void test_boundary_correction_combination(
```

Comment:
The use of `test_boundary_correction_conservation` is simple to add. The python version is going to be a pretty big chunk of very boilerplate python dispatching the long parameter set to the shorter parameter set versions -- and there will need to be one such long boilerplate python file for every combination of GH and ValenciaDivClean correction we want to test. I'm not sure whether it tests anything independent of the C++ comparison, because the separate conditions are already tested against their respective python implementations.

That said, if you think it's a useful additional check, I can do the copy-paste work -- just wanted to be clear on the not-so-pretty results before I go make the ugly python file :P.